contracts/RelayHub.sol

```diff
@@ -317,21 +318,21 @@ contract RelayHub is IRelayHub {
         (bool successPost,) = relayRequest.relayData.paymaster.call{gas:gasLimits.postRelayedCallGasLimit}(atomicData.data);
 
         if (!successPost) {
-            revertWithStatus(RelayCallStatus.PostRelayedFailed);
+            revertWithStatus(RelayCallStatus.PostRelayedFailed, atomicData.relayedCallReturnValue);
         }
 
         if (balances[relayRequest.relayData.paymaster] < atomicData.balanceBefore) {
-            revertWithStatus(RelayCallStatus.RecipientBalanceChanged);
+            revertWithStatus(RelayCallStatus.RecipientBalanceChanged, atomicData.relayedCallReturnValue);
         }
 
-        return atomicData.relayedCallSuccess ? RelayCallStatus.OK : RelayCallStatus.RelayedCallFailed;
+        return (atomicData.relayedCallSuccess ? RelayCallStatus.OK : RelayCallStatus.RelayedCallFailed, atomicData.relayedCallReturnValue);
```

Comment:
I still think the "success" check should also be applied to the returned value:
e.g.
```
if ( !atomicData.relayedCallSuccess ) {
  return (RelayCallStatus.RelayedCallFailed, atomicData.relayedCallReturnValue);
} else {
  return (RelayCallStatus.OK, "");
}
```

Reply:
I know, but that would be burning the 'relayedCallReturnValue' for idealogical reasons mostly